### DISCUSSION TITLE
Stock extruder performance

### DISCUSSION THREAD

**Discussion by mulcmu** [aegean-odyssey/mpmd_marlin_1.1.x#75] [Q&A]
What is a the expected amount of time for extruder temperature to heat up from room temperature and stabilize?  It seems like both of my machines overshoot and oscillate quite a bit more than they should.

Here was some data I collected running `mpmd_marlin_1.1.x-119r15-SM1110-ACfan-10Alimit`.  PID autotune was completed with fan at max and values loaded (`M301 P15.12 I0.76 D75.58`).  Temperature data collected at 1 second intervals over a few runs with fan at different speeds.

![image](https://user-images.githubusercontent.com/10102873/116714448-caac8780-a9a3-11eb-898c-b13dad99519c.png)

**Comment by aegean-odyssey** [comment]
In the Marlin firmware, I think that the temperature control and the thermal run-away process sometimes work at cross purposes.

When waiting on the target temperature, the firmware needs to see a stable temperature (within a few degrees) for 10s _before_  it continues on with the next G/M-code. At the same time, the thermal run-away code allows about 40s for the nozzle temperature to make progress toward the target temperature, or for a wandering nozzle temperature to return to the target temperature.

I think that the oscillations you recorded can possibly "confound" the thermal run-away code. And I agree with you: I'd expect the PID autotuning to do a better job -- certainly less ringing in the step response.

Maybe there's a refinement to the (autotuning) procedure that will produce better results. Something to look into. And perhaps someone more expert can offer some insight.
